src/main/capi/result-c.cpp

```diff
@@ -53,6 +55,11 @@ duckdb_state duckdb_translate_result(MaterializedQueryResult *result, duckdb_res
 	// zero initialize the columns (so we can cleanly delete it in case a malloc fails)
 	memset(out->__deprecated_columns, 0, sizeof(duckdb_column) * out->__deprecated_column_count);
 	for (idx_t i = 0; i < out->__deprecated_column_count; i++) {
+		BufferedSerializer serializer;
+		result->types[i].Serialize(serializer);
+		BufferedDeserializer deserializer(serializer);
+		auto cloned_type = new LogicalType(LogicalType::Deserialize(deserializer));
```

Comment:
Is there a reason for requiring a deep-copy? Wouldn't a normal copy suffice? Note that logical types are immutable, so having shared references to them should in never cause problems.

Reply:
because if I don’t deep copy it will double free

Reply:
That should not happen, logical types are copied using the regular copy constructor all over the place. Are you sure this is not masking a different problem?

Reply:
Delete the deep code logic, previously I think I miss use the column-data and result